### ISSUE TITLE
list and chart don't show data from all molecules (data limit)

### ISSUE THREAD

**Issue by gilzimmerman** [RespirationScan/issues#7]
In MS-01 the molecule list does not show all of the available molecules. Suspect this is because the molecule name is being used as the list of entries in the legend (as well as the number of plot lines) in the top right graph. Need to discover a workaround or better solution.

**Comment by rs-tamar** [comment]
Hi Gil, Following my comment in issue [#4](https://github.com/RespirationScan/issues/issues/4#issuecomment-1600676174), I have now restructured our data views structure (Postgres) in order to reduce, as much as possible, our overall view size, I also split the molecule data (name and type) from the main view to reduce it even further, in parallel to that, I also add a new counter to Power BI that represent the number of molecules behind of each selected molecule in the preset and the result is now 100% match in terms of records between our Power Bi presentation  and our database

Below are the expected calculated values per each preset:
`Acetone             |   8875|
All                 |1594081|
Aminomethanesulfonic|   6187|
Benzisothiazole     |     78|
Butane              |  44724|
Chloro              |  56547|
Dimethyl            | 101785|
Ethanol             |  20209|
Hydrogen            |  17861|
Lactic              |   7231|
Methanol            |  47891|
Sulf                |  56823|
Tetrazole           |   7373|`

**Comment by rs-tamar** [comment]
Please bear in mind that I failed to commit this latest version due to the following error:
error: failed to push some refs to 'github.com:RespirationScan/hermes.git'

**Comment by rs-tamar** [comment]
Thanks, Gil for your assistance, however, I should mention this earlier but the latest version is now been committed and the problem has been resolved

**Comment by gilzimmerman** [comment]
**Please do a git pull --rebase as I have changed the filename to the more appropriate hermes.pbix**

How did you conclude the issue was resolved? 

Did we run any tests? 
Did we have a resolution to the problem of displaying 10k different molecules on a chart? 


Take for example, patient 608.

The graph only shows 60 different molecules, which is a reasonable limit, but also nowhere near the 10,104 we would expect. 

`select count(distinct  molecule_id) 
from oldapp.compounds 
where measurement_id in (
	select measurement_id 
	from oldapp.invivomeasurements 
	where patient_id = 608 
)`

For example, there is no option to scroll past the 60th loaded molecule to show molecules 6-66 or 61-120, or anything like that.

Also, when I scroll in the Molecule list, I cannot see anything past "1'H-Cholest-2-eno[3,2-b]indole,'... 

Search does seem to show results. but we're unable to browse them. 

What do you suggest?

**Comment by rs-tamar** [comment]
Following our conversation during our stand-up meeting, I have performed further investigation 
with regard to the "Duplicate Molecule Name", however, in order that we both stay on the same line please perform the following steps

1. Click the button to clear all slicers.

2. On the Molecule Preset, select the option "All".

3. On the Patient Slicer,  select Patient# 608, Experiment#  3137, Measurement# 235 you should end up with an overall 1,840 
    Molecules (Check top counter) which represent the number across all of the records that associate with the 
    selection criteria. 

4. At this stage select the slicer - Molecule Name, you can now choose to sort the represent list based on the Molecule Name and also, choose between sorting the list descending or ascending, anyway, and by design, within this list, you will not see any duplicate Molecule Names even if technically they naturally exist behind the scene, where "duplicates" are characterised by different compounds/spectrums

**Comment by gilzimmerman** [comment]
This is likely solved by the resolution of #10. The issue seems to have disappeared, as now the full list of molecules is visible. 

Still might want to think of a better method of showing relevant molecules in the graph. Also, new issue has become evident: poor parsing has led to molecule names in quotes to be separated from molecule names outside of quotes.

**Comment by rs-tamar** [comment]
I tried to switch to #7 (after I committed #10 of course) but now I receiving the following error 

`PS C:\GitHub\PowerBI\hermes> pbi-tools compile .\hermes\ -format PBIT -overwrite True
An unhandled exception occurred.
PbiTools.Serialization.PbixProjSerializationException: The PBIX part at 'C:\GitHub\PowerBI\hermes\hermes\Version.txt' could not be deserialized and is marked as not optional.
   at PbiTools.Serialization.PowerBIPartSerializerExtensions.DeserializeSafe[T](IPowerBIPartSerializer1 serializer, Boolean isOptional) in X:\pbi-tools\BRANCHES\rc.5\src\PBI-Tools\Serialization\IPowerBIPartSerializer.cs:line 52
   at PbiTools.Model.PbixModel.FromFolder(String path) in X:\pbi-tools\BRANCHES\rc.5\src\PBI-Tools\Model\PbixModel.cs:line 202
   at PbiTools.Cli.CmdLineActions.CompilePbix(String folder, String outPath, PbiFileFormat format, Boolean overwrite) in X:\pbi-tools\BRANCHES\rc.5\src\PBI-Tools\Cli\Compile.cs:line 71
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at PowerArgs.ArgAction.Invoke()
   at PbiTools.Program.Main(String[] args) in X:\pbi-tools\BRANCHES\rc.5\src\PBI-Tools\Program.cs:line 110
PS C:\GitHub\PowerBI\hermes>`

**Comment by rs-tamar** [comment]
BTW, the file C:\GitHub\PowerBI\hermes\hermes\Version.txt doesn't exist, after the last commit the folder was left pretty much empty

**Comment by gilzimmerman** [comment]
resolved in dev release.
